esque/clients.py

```diff
@@ -1,37 +1,75 @@
 
 import click
 import confluent_kafka
 import pendulum
-from confluent_kafka import TopicPartition
+from confluent_kafka import Message
+from confluent_kafka.avro import AvroProducer
+from confluent_kafka.cimpl import KafkaError
 
-from esque.config import Config, PING_GROUP_ID, PING_TOPIC
+from esque.avromessage import AvroFileReader, AvroFileWriter
+from esque.config import Config
 from esque.errors import raise_for_kafka_error, raise_for_message
 from esque.helpers import delivery_callback, delta_t
+from esque.message import KafkaMessage, PlainTextFileReader, PlainTextFileWriter, FileReader, FileWriter
+from esque.schemaregistry import SchemaRegistryClient
+from abc import ABC, abstractmethod
 
-DEFAULT_RETENTION_MS = 7 * 24 * 60 * 60 * 1000
 
+class AbstractConsumer(ABC):
+    def __init__(self, group_id: str, topic_name: str, last: bool):
+        offset_reset = "earliest"
+        if last:
+            offset_reset = "latest"
 
-class Consumer:
-    def __init__(self):
         self._config = Config().create_confluent_config()
         self._config.update(
             {
-                "group.id": PING_GROUP_ID,
+                "group.id": group_id,
                 "error_cb": raise_for_kafka_error,
                 # We need to commit offsets manually once we"re sure it got saved
                 # to the sink
                 "enable.auto.commit": True,
                 "enable.partition.eof": False,
                 # We need this to start at the last committed offset instead of the
                 # latest when subscribing for the first time
-                "default.topic.config": {"auto.offset.reset": "latest"},
+                "default.topic.config": {"auto.offset.reset": offset_reset},
             }
         )
         self._consumer = confluent_kafka.Consumer(self._config)
-        self._assign_exact_partitions(PING_TOPIC)
+        self._subscribe(topic_name)
+
+    def _subscribe(self, topic: str) -> None:
+        self._consumer.subscribe([topic])
+
+    @abstractmethod
+    def consume(self, amount: int) -> int:
+        pass
+
+    def _consume_single_message(self) -> Optional[Message]:
+        poll_limit = 10
+        counter = 0
+        while counter < poll_limit:
+            message = self._consumer.poll(timeout=10)
+            if message is None:
+                counter += 1
+                continue
+            if message.error() is not None:
+                if message.error().code() == KafkaError._PARTITION_EOF:
+                    print("\nEnd of partition reached!".format(**locals()))
+                    break
+                else:
+                    raise RuntimeError(message.error().str())
+            raise_for_message(message)
+            return message
+
 
-    def consume_ping(self) -> Optional[Tuple[str, int]]:
+class PingConsumer(AbstractConsumer):
+    def consume(self, amount: int) -> Optional[Tuple[str, int]]:
         msg = self._consumer.consume(timeout=10)[0]
```

Comment:
You should use `self._consume_single_message` here. Then you can also get rid of `raise_for_message(msg)` which is already called there.